src/Build/BackEnd/Components/RequestBuilder/RequestBuilder.cs

```diff
@@ -1162,9 +1167,6 @@ private async Task<BuildResult> BuildProject()
 
                 _requestEntry.Request.BuildEventContext = _projectLoggingContext.BuildEventContext;
 
-                // Determine the set of targets we need to build
-                (string name, TargetBuiltReason reason)[] allTargets = _requestEntry.RequestConfiguration
```

Comment:
note: this PR changes the order of 
 `_requestEntry.RequestConfiguration.RetrieveFromCache();`
`(string name, TargetBuiltReason reason)[] allTargets = _requestEntry.RequestConfiguration
   .GetTargetsUsedToBuildRequest(_requestEntry.Request).ToArray();`
from looking at the code I'm pretty sure these don't interact in any way, but logging this if we eventually run into some side effect downstream of this

Reply:
Add this comment as PR notes